src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -31,19 +32,35 @@ public void init() {
   }
 
   @Override
-  // Get a specific user. Return null if not found. TODO: User Validation
+  // Get a specific user. Return null if not found.
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String userGithub = request.getParameter("githubID");
     Key userKey = KeyFactory.createKey(PersonBuilder.ENTITY_NAME, userGithub);
-    System.out.println(userKey);
 
+    Entity userEntity;
+    try {
+      userEntity = datastore.get(userKey);
+    } catch (Exception e) {
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage() + "/n/n" + e.getStackTrace(),
+          HttpServletResponse.SC_BAD_REQUEST,
+          "The user requested could not be found on the server."
+              + "Please ensure that the user has an account with us.");
+      return;
+    }
     PersonObject userObject;
     try {
-      userObject = new PersonBuilder().buildPersonObject(datastore.get(userKey));
-      System.out.println(userObject);
+      userObject = new PersonBuilder().buildPersonObject(userEntity);
     } catch (Exception e) {
```

Comment:
This looks good, although I'm wondering if we could move this functionality to a public static function like `getPersonObjectById(id)` that we could use in other places like the `hasProfile` check in the auth servlet?

Reply:
will do!